packages/react-environment/src/MixerProvider.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { ReactNode, FC, useState, useEffect, useCallback } from 'react';
+import { Mixer, MixerAssetGroup } from '@webb-tools/sdk-mixer';
+import { useApi } from '@webb-dapp/react-hooks';
+import { notification } from '@webb-dapp/ui-components';
+import { getTokenName } from '@webb-dapp/react-components';
+
+export interface MixerContextData {
+  init: () => Promise<Mixer>;
+  mixerGroups: MixerAssetGroup[];
+}
+
+// ensure that mixer always exist
+export const MixerContext = React.createContext<MixerContextData>({} as MixerContextData);
+
+interface Props {
+  children: ReactNode;
+}
+
+/**
+ * @name MixerProvider
+ * @description context provider to support mixer.
+ */
+export const MixerProvider: FC<Props> = ({ children }) => {
+  const { api, connected } = useApi();
+  const [mixerGroups, setMixerGroups] = useState<MixerAssetGroup[]>([]);
+
+  useEffect(() => {
+    if (!connected) return;
+    try {
+      api.query.mixer.mixerGroups
+        .entries()
+        .toPromise()
+        .then((result) => {
```

Comment:
@drewstone this call never resolves (on the anon local node)! 
any ideas?

Reply:
Have you tried using `useCall`? https://github.com/webb-tools/webb-dapp/blob/master/packages/react-hooks/src/useCall.ts
```
  const mixerGroups = useCall<MixerGroup[]>('query.mixer.mixerGroups, []);
```